foxpuppet/windows/browser/notifications.py

```diff
@@ -0,0 +1,114 @@
+            return self.root.get_attribute('label')
+
+    @property
+    def origin(self):
+        """Provide access to the notification origin.
+
+        :returns: The notification origin.
+        """
+        with self.selenium.context('chrome'):
+            return self.root.get_attribute('origin')
+
+    def close(self):
+        """Close the notification."""
+        with self.selenium.context('chrome'):
+            self.root.find_anonymous_element_by_attribute(
+                'anonid', 'closebutton').click()
+        self.window.wait_for_notification(None)
+
+
+class AddOnInstallBlocked(BaseNotification):
+    """Add-on install blocked notification."""
+
+    def allow(self):
+        """Allow the add-on to be installed"""
+        with self.selenium.context('chrome'):
+            self.root.find_anonymous_element_by_attribute('anonid', 'button') \
+                .find_anonymous_element_by_attribute('anonid', 'button') \
+                .click()
+
+
+class AddOnInstallConfirmation(BaseNotification):
+    """Add-on install confirmation notification."""
+
+    _addon_name_locator = (
+        By.CSS_SELECTOR, '#addon-install-confirmation-content label')
+    _cancel_locator = (By.ID, 'addon-install-confirmation-cancel')
+    _confirm_locator = (By.ID, 'addon-install-confirmation-accept')
+
+    @property
+    def addon_name(self):
+        """Provide access to the add-on name.
+
+        :returns: The add-on name.
+        """
+        with self.selenium.context('chrome'):
+            label = self.root.find_element(*self._addon_name_locator)
+            return label.get_attribute('value')
+
+    def cancel(self):
+        """Cancel add-on install."""
+        with self.selenium.context('chrome'):
+            self.root.find_element(*self._cancel_locator).click()
+
+    def install(self):
+        """Confirm add-on install."""
+        with self.selenium.context('chrome'):
+            self.root.find_element(*self._confirm_locator).click()
+
+
+class AddOnInstallComplete(BaseNotification):
+    """Add-on install complete notification."""
+
+
+class AddOnInstallRestart(BaseNotification):
+    """Add-on install restart notification."""
+
+
+class AddOnInstallFailed(BaseNotification):
+    """Add-on install failed notification."""
+
+
+class AddOnProgress(BaseNotification):
+    """Add-on progress notification."""
+
+
+class NotificationFactory():
+    NOTIFICATIONS = {
+        'addon-install-blocked-notification': AddOnInstallBlocked,
+        'addon-install-confirmation-notification': AddOnInstallConfirmation,
+        'addon-install-complete-notification': AddOnInstallComplete,
+        'addon-install-restart-notification': AddOnInstallRestart,
+        'addon-install-failed-notification': AddOnInstallFailed,
+        'addon-progress-notification': AddOnProgress,
+    }
+
+    def create(self, window, root):
+        _id = root.get_attribute('id')
+        return self.NOTIFICATIONS.get(_id, BaseNotification)(window, root)
```

Comment:
I would propose that you move this create method into the base notification class and make it static. That way we can simply do `notification = Notification.create(window, element)`. Also the available notification types would be accessible directly without having to use a different class.